subsys/net/lib/nrf_cloud/src/nrf_cloud_agnss.c

```diff
@@ -465,6 +465,13 @@ static size_t get_next_agnss_element(struct nrf_cloud_agnss_element *element,
 	 * each element.
 	 */
 	if (elements_left_to_process == 0) {
+		/* If there's no more data, reset the parser state. */
+		if (buf_len == 0) {
+			elements_left_to_process = 0;
```

Comment:
This line is redundant -- the if statement on line 467 cannot be entered unless elements_left_to_process == 0 already.

Reply:
Ah, yes. I was in a bit of a hurry. The main point is the change on line 606, it's the only change that is actually needed. Also, setting the `element_type` to `NRF_CLOUD_AGNSS__TYPE_INVALID` is not mandatory to fix the problem I found, but I thought it would be nice to re-initialize both values. This is the second similar problem I have ran into and setting `element_type` to `NRF_CLOUD_AGNSS__TYPE_INVALID` should prevent all similar problems and make the parsing error visible.